### DISCUSSION TITLE
Bug located in ... changing offset size for multiple bodies in stack direction for hex meshing

### DISCUSSION THREAD

**Discussion by ragulselvaraj20** [ansys/pyprimemesh#670] [Q&A] [answered]
### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

In stack direction, there are 5 bodies are there. All are having different thickness in the stack direction. During the hex mesh generation, keeping the same maximum offset distance in the stack direction leading to high mesh count. In order to reduce the mesh count, offset distance need to be changed.

### 📝 Steps to reproduce

How to change offset distance in the stack direction for multiple bodies.

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

ANSYS 2023R2

### 🐍 Which Python version are you using?

3.8

### 📦 Installed packages

```shell
pyprimemesh
```

**Comment by agvarghe** [comment, marked as answer]
Hi @ragulselvaraj20 

I would recommend using size controls to control the number of layers on each body , this would provide a control and reduce the mesh count. 

We have added an example to demonstrate this scenario for Volume Sweep feature . The example shows how to use edge sizing to define the number of cells through each layer. A named selection is defined in the scdoc for the edges and is named such that the length of the edges is included in the name. This is then used to define the number of cells through each volume.

Please do find the link to the example below 

https://prime.docs.pyansys.com/version/dev/examples/gallery_examples/gallery/09_multi_layer_quad_mesh_pcb.html#sphx-glr-examples-gallery-examples-gallery-09-multi-layer-quad-mesh-pcb-py

CC: @ninad-kamat , @waltersma

**Comment by waltersma** [comment]
Hi @ragulselvaraj20, 

If it is ok with you I will move this to the discussions and class it as question answered?  Please let me know if this is not the case and we can re open. (Cc @agvarghe)

Thanks!

**Comment by ragulselvaraj20** [comment]
Thanks @agvarghe. Is it working in 2023 R2 version?

**Comment by agvarghe** [comment]
HI @ragulselvaraj20 
This feature should work but the example mentioned wouldnt work since its prepared for 2024R1.
